libs/shared/design-system/src/lib/tree/tree.model.ts

```diff
@@ -0,0 +1,12 @@
+import { PropsWithChildren, ReactNode } from 'react';
+
+export type TIcon = ReactNode | { icon: ReactNode; position: 'before' | 'after' }[];
+
+export type TTree = PropsWithChildren<{
+  title: string | ReactNode;
+  className?: string;
+  icon?: TIcon;
+  collapsable?: boolean;
+}>;
```

Comment:
It is better to use interfaces whenever possible
```suggestion
export interface ITree extends PropsWithChildren {
  title: string | ReactNode;
  className?: string;
  icon?: TIcon;
  collapsable?: boolean;
}
```

Reply:
it depends. I don't think we will get to the point that performance related to difference in interfaces and types will make difference. Interfaces has its own problems (you can create two interfaces with the same name and won't get error). In this specific case I think it is more important that performance (which may be important but in very big apps)